src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseInsecureDeserializerJsonNetWithoutBinder.cs

```diff
@@ -0,0 +1,250 @@
+                        valueContentAbstractValue,
+                        (object o) => o is int i && i != 0),   // None is 0, and anything other than None is flagged.
+                TypeNameHandlingIndex),
+            new PropertyMapper(
+                "Binder",
+                PropertySetCallbacks.FlagIfNull,
+                SerializationBinderIndex),      // Binder & SerializationBinder have the same underlying value.
+            new PropertyMapper(
+                "SerializationBinder",
+                PropertySetCallbacks.FlagIfNull,
+                SerializationBinderIndex));     // Binder & SerializationBinder have the same underlying value.
+
+        private static readonly HazardousUsageEvaluatorCollection HazardousUsageEvaluators = new HazardousUsageEvaluatorCollection(
+            SecurityHelpers.JsonSerializerDeserializationMethods.Select(
+                (string methodName) => new HazardousUsageEvaluator(
+                    methodName,
+                    PropertySetCallbacks.HazardousIfAllFlaggedAndAtLeastOneKnown))
+                .Concat(
+                    new HazardousUsageEvaluator(
+                        HazardousUsageEvaluatorKind.Initialization,
+                        PropertySetCallbacks.HazardousIfAllFlaggedAndAtLeastOneKnown)));
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    WellKnownTypeProvider wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilationStartAnalysisContext.Compilation);
+                    if (!wellKnownTypeProvider.TryGetTypeByMetadataName(
+                            WellKnownTypeNames.NewtonsoftJsonJsonSerializer,
+                            out INamedTypeSymbol jsonSerializerSymbol))
+                    {
+                        return;
+                    }
+
+                    PooledHashSet<(IOperation Operation, ISymbol ContainingSymbol)> rootOperationsNeedingAnalysis = PooledHashSet<(IOperation, ISymbol)>.GetInstance();
+
+                    compilationStartAnalysisContext.RegisterOperationBlockStartAction(
+                        (OperationBlockStartAnalysisContext operationBlockStartAnalysisContext) =>
+                        {
+                            ISymbol owningSymbol = operationBlockStartAnalysisContext.OwningSymbol;
+
+                            // TODO: Handle case when exactly one of the below rules is configured to skip analysis.
+                            if (owningSymbol.IsConfiguredToSkipAnalysis(
```

Comment:
You should probably also update the `Configurable Rules` list [here](https://github.com/dotnet/roslyn-analyzers/blob/master/docs/Analyzer%20Configuration.md#excluded-symbol-names)